OAuth: Support mapping generic OAuth users to Grafana organizations (#23661)
grafana/grafana: docs/sources/auth/generic-oauth.md

To ease configuration of a proper JMESPath expression, you can test/evaluate expressions with custom payloads at http://jmespath.org/.

> ### Role and Organization mapping

Reviewer: ```suggestion
### Role and organization mapping
```

Reviewer: Headings should be [sentence case](https://github.com/grafana/grafana/blob/master/contribute/style-guides/documentation-style-guide.md#headings).

> Defining role and organization mapping is handled by setting a path to TOML mapping file via `group_mappings_file`, example:

Reviewer: ```suggestion
Defining role and organization mapping is handled by setting a path to a TOML mapping file via `group_mappings_file`, for example:
```

> ```bash

Reviewer: ```suggestion
```ini
```

[auth.generic_oauth]
...
group_mappings_file = /etc/grafana/oauth-mappings.toml
```

**Basic example:**

> In the following example user will get `Editor` as role in OrgID `2` when authenticating. The value of the property `role` will be the resulting role if the role is a proper Grafana role, i.e. `Viewer`, `Editor` or `Admin`.

Reviewer: ```suggestion
In the following example, the user will get `Editor` as role in OrgID `2` when authenticating. The value of the property `role` will be the resulting role if the role is a proper Grafana role, i.e. `Viewer`, `Editor` or `Admin`.
```

Config:

> ```bash

Reviewer: ```suggestion
```toml
```

[[group_mappings]]
role_attribute_path = "role"
org_id = 2
```

**Advanced example:**

> In the following example user will get `Admin` as role in Orgs `1` and `2` when authenticating since it has a group `admin`. If a user has a group `editor` it will get `Editor` as role for Org `1`, otherwise `Viewer` in Org `2`.

Reviewer: The original wording doesn't really make sense to me, I try to make sense of it through the following wording:

```suggestion
In the following example, the user will get `Admin` as role in organizations `1` and `2` when authenticating if they have a group `admin`. If, on the other hand, the user has a group `editor`, they will get `Editor` as role for organization `1. If the user isn't in the `admin` group, they will get the `Viewer` role in organization `2`.
```

Config:

> ```bash

Reviewer: ```suggestion
```toml
```
